libmuscle/python/libmuscle/manager/deadlock_detector.py

```diff
@@ -13,11 +13,11 @@ class DeadlockDetector(Thread):
     """The DeadlockDetector attempts to detect when multiple instances are stuck waiting
     for each other.
 
-    This class is responsible for handling WAITING_FOR_RECEIVE and
+    This class is responsible for handling WAITING_FOR_RECEIVE, IS_DEADLOCKED and
     WAITING_FOR_RECEIVE_DONE MMP messages, which are submitted by the MMPServer.
 
     When a deadlock is detected, the cycle of instances that is waiting on each other is
-    logged with FATAL severity. If this deadlock does not get resoled in
+    logged with FATAL severity. If this deadlock does not get resolved in
```

Comment:
Actually, it's only logged if it isn't resolved, right? That's how it should be, so the comment should be clarified I think.

Reply:
This is currently logged as soon as the manager identifies a cycle of waiting instances. I think this will never be resolved, but I'm not sure.

I can also update the logic to only print the deadlock cycle when any of the deadlocked instances calls `is_deadlocked` and starts shutting down. At that point we're sure that there was a deadlock and that the simulation will shut down.